### ISSUE TITLE
RuntimeError: No available kernel. Aborting execution.

### ISSUE THREAD

**Issue by TPF0628** [facebookresearch/segment-anything-2#57]
How to fix it on windows ?

**Comment by Pizzawookiee** [comment]
See closed issue #48

**Comment by ronghanghu** [comment]
Hi, we have added [`INSTALL.md`](https://github.com/facebookresearch/segment-anything-2/blob/main/INSTALL.md) as an installation FAQ.

Regarding this error:

This is probably because your machine doesn't have a GPU or a compatible PyTorch version for Flash Attention (see also https://discuss.pytorch.org/t/using-f-scaled-dot-product-attention-gives-the-error-runtimeerror-no-available-kernel-aborting-execution/180900 for a discussion in PyTorch forum). You may be able to resolve this error by replacing the line
```python
OLD_GPU, USE_FLASH_ATTN, MATH_KERNEL_ON = get_sdpa_settings()
```
in [`sam2/modeling/sam/transformer.py`](sam2/modeling/sam/transformer.py) with
```python
OLD_GPU, USE_FLASH_ATTN, MATH_KERNEL_ON = True, True, True
```
to relax the attention kernel setting and use other kernels than Flash Attention.
